tests/integration_tests/bugs/test_gh586.py

```diff
@@ -0,0 +1,37 @@
+"""Integration test for pull #586
+
+If a non-default AuthorizedKeysFile is specified in /etc/ssh/sshd_config,
+ensure we can still ssh as expected.
+"""
+import paramiko
+import pytest
+from io import StringIO
+from tests.integration_tests.assets import get_test_keypair
+
+
+public_key, private_key = get_test_keypair()
+USER_DATA = """\
+#cloud-config
+bootcmd:
+  - sed -i 's/#AuthorizedKeysFile.*/AuthorizedKeysFile\\ .ssh\\/authorized_keys2/' /etc/ssh/sshd_config
+ssh_authorized_keys:
+ - {public_key}
+""".format(public_key=public_key)  # noqa: E501
+
+
+@pytest.mark.sru_2020_11
+@pytest.mark.user_data(USER_DATA)
+def test_non_default_authorized_keys(client):
+    sshd = client.read_from_file('/etc/ssh/sshd_config')
+    assert 'AuthorizedKeysFile .ssh/authorized_keys2' in sshd
+
+    ssh_dir = client.execute('ls /home/ubuntu/.ssh').stdout
+    assert 'authorized_keys2' in ssh_dir
```

Comment:
Should we assert the absence of the regular config line/files here?

Reply:
I'm adding a check for sshd_config, but the original authorized_keys file actually still exists from the image snapshot. It doesn't affect this test, but I wonder if that means we should we doing more than just a `cloud-init clean --logs` when creating an image snapshot.

Reply:
Yeah, I think we probably should be. Not only do we have that file still around, we also have the default user already created. (When we decide what else we want to do, we should consider expanding `clean` to perform it for us so it isn't specific to the tests.)

This is what the cloud_tests do, so not much to be learned from that: https://github.com/canonical/cloud-init/blob/bd76d5cfbde9c0801cefa851db82887e0bab34c1/tests/cloud_tests/releases.yaml#L14-L17